pgjdbc/src/main/java/org/postgresql/jdbc/ArrayDecoding.java

```diff
@@ -471,6 +475,12 @@ Object parseValue(String stringVal, BaseConnection connection) throws SQLExcepti
       throw org.postgresql.Driver.notImplemented(PgArray.class, "readArray(data,oid)");
     }
 
+    // 42.2.x should return enums as strings
+    int type = connection.getTypeInfo().getSQLType(typeName);
+    if (type == Types.CHAR || type == Types.VARCHAR) {
```

Comment:
It appears that enum[] comes back as VARCHAR. Is CHAR actually needed here, or just VARCHAR?

Reply:
I copied the condition from the previous code to minimize the chances for other regressions.

Reply:
good enough